Tidied up reference landing page introduction (#7715)
GoogleChrome/developer.chrome.com: site/en/docs/extensions/reference/index.md

layout: 'layouts/reference-landing.njk'
---

Most extensions need access to one or more Chrome Extensions APIs to function. This API reference describes the APIs available for use in extensions and presents example use cases.

## Common Extensions API features

> An Extensions API typically consists of a namespace containing manifest fields that must be included in your extension's `manifest.json` file to use that API. Those fields can include objects within `manifest.json,` such as the action object required by the `chrome.action` API, as well as [permissions](/docs/extensions/mv3/declare_permissions/) declarations for APIs that require permissions.

Reviewer: ```suggestion
An Extensions API consists of a namespace containing methods and properties for doing extensions work, and usually, but not always, manifest fields for the `manifest.json` file. For example, the `chrome.action` namespace requires an `"action"` object in the manifest. Additionally, many APIs require [permissions](/docs/extensions/mv3/declare_permissions/) in the manifest.
```

> Methods in `chrome.*` APIs are **asynchronous** unless stated otherwise. Asynchronous methods return immediately, without waiting for the operation that calls them to finish. Use [promises](/docs/extensions/mv3/promises/) to get the results of these methods. For more information, see [Asynchronous methods](/docs/extensions/mv3/architecture-overview/#async-sync).

Reviewer: ```suggestion
Methods in extension APIs are **asynchronous** unless stated otherwise. Asynchronous methods return immediately, without waiting for the operation that calls them to finish. Use [promises](/docs/extensions/mv3/promises/) to get the results of these methods. For more information, see [Asynchronous methods](/docs/extensions/mv3/architecture-overview/#async-sync).
```

> <!--What else are users likely to need to know about early? Should I bring back the mention of callbacks in the asynchronous methods explanation? And should I be able to edit the table structures in the generated content?-->

Reviewer: ```suggestion
```

Reviewer: >What else are users likely to need to know about early?

Ask around, but let's talk about anything suggested.

>Should I bring back the mention of callbacks in the asynchronous methods explanation?

We're in middle of a change from callbacks to promises, and we're close to finishing. We'll remove this paragraph eventually.

>And should I be able to edit the table structures in the generated content?

The tables aren't in the source, they're created by the generation process. I'm working changing this.
